src/main/java/ljv/GraphBuilder.java

```diff
@@ -0,0 +1,191 @@
+package ljv;
+
+import java.lang.reflect.*;
+import java.util.*;
+
+public class GraphBuilder {
+    private final IdentityHashMap<Object, String> objectsId = new IdentityHashMap<>();
+    private final ObjSettings oSettings;
+    private final StringBuilder out = new StringBuilder();
+    private final LJV ljv;
+
+    public GraphBuilder(LJV ljv) {
+        this.ljv = ljv;
+        this.oSettings = new ObjSettings(ljv);
+    }
+
+    private String dotName(Object obj) {
+        return obj == null ? "NULL" : objectsId.computeIfAbsent(obj, s -> "n" + (objectsId.size() + 1));
+    }
+
+    private void processPrimitiveArray(Object obj) {
+        out.append("\t")
+                .append(dotName(obj))
+                .append("[label=<\n")
+                .append("\t\t<table border='0' cellborder='1' cellspacing='0'>\n")
+                .append("\t\t\t<tr>\n");
+        for (int i = 0, len = Array.getLength(obj); i < len; i++) {
+            out.append("\t\t\t\t<td>")
+                .append(Quote.quote(String.valueOf(Array.get(obj, i))))
+                .append("</td>\n");
+        }
+        out.append("\t\t\t</tr>\n\t\t</table>\n\t>];\n");
+    }
+
+    private void processObjectArray(Object obj) {
+        out.append("\t")
+                .append(dotName(obj))
+                .append("[label=<\n")
+                .append("\t\t<table border='0' cellborder='1' cellspacing='0' cellpadding='9'>\n")
+                .append("\t\t\t<tr>\n");
+        int len = Array.getLength(obj);
+        for (int i = 0; i < len; i++) {
+            out.append("\t\t\t\t<td port=\"f").append(i).append("\"></td>\n");
+        }
+        out.append("\t\t\t</tr>\n\t\t</table>\n\t>];\n");
+        for (int i = 0; i < len; i++) {
+            Object ref = Array.get(obj, i);
+            if (ref == null)
+                continue;
+            generateDotInternal(ref);
+            out.append("\t")
+                    .append(dotName(obj))
+                    .append(":f")
+                    .append(i)
+                    .append(" -> ")
+                    .append(dotName(ref))
+                    .append("[label=\"")
+                    .append(i)
+                    .append("\",fontsize=12];\n");
+        }
+    }
+
+    private int getFieldSize(LJV ljv, Object obj, Field[] fs) {
```

Comment:
You don't need `LJV` as an argument here, it's already available as a class member